Malformed Config Files Kill Server With Useless Error Message [2.7.0]

### Play Version (2.5.x / etc)

2.7.0

### API (Scala / Java / Neither / Both)

Scala

### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)

Linux 4.15.0-45-generic #48-Ubuntu SMP Tue Jan 29 16:28:13 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)

openjdk version "1.8.0_191"
OpenJDK Runtime Environment (build 1.8.0_191-8u191-b12-2ubuntu0.18.04.1-b12)
OpenJDK 64-Bit Server VM (build 25.191-b12, mixed mode)

### Library Dependencies

N/A

### Expected Behavior

Please describe the expected behavior of the issue, starting from the first action.

1. Malformed application.conf is placed in ./conf
2. Run Play dev-mode
3. Attempt to load a page
4. Error: Malformed application.conf is logged/reported.

### Actual Behavior

Please provide a description of what actually happens, working from the same starting point.

Be descriptive: "it doesn't work" does not describe what the behavior actually is -- instead, say "the page renders a 500 error code with no body content."  Copy and paste logs, and include any URLs.  Turn on internal Play logging with `<logger name="play" value="TRACE"/>` if there is no log output.  

* Error 500 is provided to the browser with no body.
* At step 4 above, the following output has been provided to the log (with "TRACE" enabled as described above), producing:  
```console
--- (Running the application, auto-reloading is enabled) ---

[info] p.c.s.AkkaHttpServer - Listening for HTTP on /0.0.0.0:9000

(Server started, use Enter to stop and go back to the console...)

[error] a.a.ActorSystemImpl - Internal server error, sending 500 response
akka.http.impl.util.One2OneBidiFlow$OutputTruncationException: Inner flow was completed without producing result elements for 1 outstanding elements
	at akka.http.impl.util.One2OneBidiFlow$OutputTruncationException$.apply(One2OneBidiFlow.scala:22)
	at akka.http.impl.util.One2OneBidiFlow$OutputTruncationException$.apply(One2OneBidiFlow.scala:22)
	at akka.http.impl.util.One2OneBidiFlow$One2OneBidi$$anon$1$$anon$4.onUpstreamFinish(One2OneBidiFlow.scala:97)
	at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:504)
	at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:378)
	at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:588)
	at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:472)
	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:563)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:745)
	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:760)
```

### Reproducible Test Case

Sample project available here:
https://github.com/SattaiLanfear/play27malformedconf/

This project was generated using the play scala giter8 on 2/10/2019.
application.conf and logback.xml are the only files that should have been changed from default.

When running the provided example with the [netty backend](https://www.playframework.com/documentation/2.7.x/NettyServer#Netty-Server-Backend) I get much better error message:

<details>
<summary>Click here to view the stack trace</summary>

```
[error] p.c.s.n.PlayRequestHandler - Exception caught in Netty
play.api.Configuration$$anon$1: Configuration error[application.conf @ file:./play27malformedconf/target/scala-2.12/classes/application.conf: 3: Expecting a value but got wrong token: '?' (Reserved character '?' is not allowed outside quotes) (if you intended '?' (Reserved character '?' is not allowed outside quotes) to be part of a key or string value, try enclosing the key or value in double quotes, or you may be able to rename the file .properties rather than .conf)]
	at play.api.Configuration$.configError(Configuration.scala:171)
	at play.api.Configuration$.load(Configuration.scala:109)
	at play.api.Configuration$.load(Configuration.scala:121)
	at play.api.Configuration$.load(Configuration.scala:128)
	at play.api.http.DefaultHttpErrorHandler$.setEditor$lzycompute(HttpErrorHandler.scala:487)
	at play.api.http.DefaultHttpErrorHandler$.setEditor(HttpErrorHandler.scala:486)
	at play.api.http.DefaultHttpErrorHandler$.onServerError(HttpErrorHandler.scala:495)
	at play.core.server.Server$$anonfun$handleErrors$1$1.applyOrElse(Server.scala:104)
	at play.core.server.Server$$anonfun$handleErrors$1$1.applyOrElse(Server.scala:100)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
Caused by: com.typesafe.config.ConfigException$Parse: application.conf @ file:./play27malformedconf/target/scala-2.12/classes/application.conf: 3: Expecting a value but got wrong token: '?' (Reserved character '?' is not allowed outside quotes) (if you intended '?' (Reserved character '?' is not allowed outside quotes) to be part of a key or string value, try enclosing the key or value in double quotes, or you may be able to rename the file .properties rather than .conf)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:201)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:197)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseValue(ConfigDocumentParser.java:251)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseObject(ConfigDocumentParser.java:475)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parse(ConfigDocumentParser.java:648)
	at com.typesafe.config.impl.ConfigDocumentParser.parse(ConfigDocumentParser.java:14)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:260)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:248)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:180)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:174)
[error] p.c.s.n.PlayRequestHandler - Exception caught in Netty
play.api.Configuration$$anon$1: Configuration error[application.conf @ file:./play27malformedconf/target/scala-2.12/classes/application.conf: 3: Expecting a value but got wrong token: '?' (Reserved character '?' is not allowed outside quotes) (if you intended '?' (Reserved character '?' is not allowed outside quotes) to be part of a key or string value, try enclosing the key or value in double quotes, or you may be able to rename the file .properties rather than .conf)]
	at play.api.Configuration$.configError(Configuration.scala:171)
	at play.api.Configuration$.load(Configuration.scala:109)
	at play.api.Configuration$.load(Configuration.scala:121)
	at play.api.Configuration$.load(Configuration.scala:128)
	at play.api.http.DefaultHttpErrorHandler$.setEditor$lzycompute(HttpErrorHandler.scala:487)
	at play.api.http.DefaultHttpErrorHandler$.setEditor(HttpErrorHandler.scala:486)
	at play.api.http.DefaultHttpErrorHandler$.onServerError(HttpErrorHandler.scala:495)
	at play.core.server.Server$$anonfun$handleErrors$1$1.applyOrElse(Server.scala:104)
	at play.core.server.Server$$anonfun$handleErrors$1$1.applyOrElse(Server.scala:100)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
Caused by: com.typesafe.config.ConfigException$Parse: application.conf @ file:./play27malformedconf/target/scala-2.12/classes/application.conf: 3: Expecting a value but got wrong token: '?' (Reserved character '?' is not allowed outside quotes) (if you intended '?' (Reserved character '?' is not allowed outside quotes) to be part of a key or string value, try enclosing the key or value in double quotes, or you may be able to rename the file .properties rather than .conf)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:201)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:197)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseValue(ConfigDocumentParser.java:251)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseObject(ConfigDocumentParser.java:475)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parse(ConfigDocumentParser.java:648)
	at com.typesafe.config.impl.ConfigDocumentParser.parse(ConfigDocumentParser.java:14)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:260)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:248)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:180)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:174)
[error] p.c.s.n.PlayRequestHandler - Exception caught in Netty
play.api.Configuration$$anon$1: Configuration error[application.conf @ file:./play27malformedconf/target/scala-2.12/classes/application.conf: 3: Expecting a value but got wrong token: '?' (Reserved character '?' is not allowed outside quotes) (if you intended '?' (Reserved character '?' is not allowed outside quotes) to be part of a key or string value, try enclosing the key or value in double quotes, or you may be able to rename the file .properties rather than .conf)]
	at play.api.Configuration$.configError(Configuration.scala:171)
	at play.api.Configuration$.load(Configuration.scala:109)
	at play.api.Configuration$.load(Configuration.scala:121)
	at play.api.Configuration$.load(Configuration.scala:128)
	at play.api.http.DefaultHttpErrorHandler$.setEditor$lzycompute(HttpErrorHandler.scala:487)
	at play.api.http.DefaultHttpErrorHandler$.setEditor(HttpErrorHandler.scala:486)
	at play.api.http.DefaultHttpErrorHandler$.onServerError(HttpErrorHandler.scala:495)
	at play.core.server.Server$$anonfun$handleErrors$1$1.applyOrElse(Server.scala:104)
	at play.core.server.Server$$anonfun$handleErrors$1$1.applyOrElse(Server.scala:100)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
Caused by: com.typesafe.config.ConfigException$Parse: application.conf @ file:./play27malformedconf/target/scala-2.12/classes/application.conf: 3: Expecting a value but got wrong token: '?' (Reserved character '?' is not allowed outside quotes) (if you intended '?' (Reserved character '?' is not allowed outside quotes) to be part of a key or string value, try enclosing the key or value in double quotes, or you may be able to rename the file .properties rather than .conf)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:201)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:197)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseValue(ConfigDocumentParser.java:251)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseObject(ConfigDocumentParser.java:475)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parse(ConfigDocumentParser.java:648)
	at com.typesafe.config.impl.ConfigDocumentParser.parse(ConfigDocumentParser.java:14)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:260)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:248)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:180)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:174)
[error] p.c.s.n.PlayRequestHandler - Exception caught in Netty
play.api.Configuration$$anon$1: Configuration error[application.conf @ file:./play27malformedconf/target/scala-2.12/classes/application.conf: 3: Expecting a value but got wrong token: '?' (Reserved character '?' is not allowed outside quotes) (if you intended '?' (Reserved character '?' is not allowed outside quotes) to be part of a key or string value, try enclosing the key or value in double quotes, or you may be able to rename the file .properties rather than .conf)]
	at play.api.Configuration$.configError(Configuration.scala:171)
	at play.api.Configuration$.load(Configuration.scala:109)
	at play.api.Configuration$.load(Configuration.scala:121)
	at play.api.Configuration$.load(Configuration.scala:128)
	at play.api.http.DefaultHttpErrorHandler$.setEditor$lzycompute(HttpErrorHandler.scala:487)
	at play.api.http.DefaultHttpErrorHandler$.setEditor(HttpErrorHandler.scala:486)
	at play.api.http.DefaultHttpErrorHandler$.onServerError(HttpErrorHandler.scala:495)
	at play.core.server.Server$$anonfun$handleErrors$1$1.applyOrElse(Server.scala:104)
	at play.core.server.Server$$anonfun$handleErrors$1$1.applyOrElse(Server.scala:100)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
Caused by: com.typesafe.config.ConfigException$Parse: application.conf @ file:./play27malformedconf/target/scala-2.12/classes/application.conf: 3: Expecting a value but got wrong token: '?' (Reserved character '?' is not allowed outside quotes) (if you intended '?' (Reserved character '?' is not allowed outside quotes) to be part of a key or string value, try enclosing the key or value in double quotes, or you may be able to rename the file .properties rather than .conf)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:201)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:197)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseValue(ConfigDocumentParser.java:251)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseObject(ConfigDocumentParser.java:475)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parse(ConfigDocumentParser.java:648)
	at com.typesafe.config.impl.ConfigDocumentParser.parse(ConfigDocumentParser.java:14)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:260)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:248)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:180)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:174)
[error] p.c.s.n.PlayRequestHandler - Exception caught in Netty
play.api.Configuration$$anon$1: Configuration error[application.conf @ file:./play27malformedconf/target/scala-2.12/classes/application.conf: 3: Expecting a value but got wrong token: '?' (Reserved character '?' is not allowed outside quotes) (if you intended '?' (Reserved character '?' is not allowed outside quotes) to be part of a key or string value, try enclosing the key or value in double quotes, or you may be able to rename the file .properties rather than .conf)]
	at play.api.Configuration$.configError(Configuration.scala:171)
	at play.api.Configuration$.load(Configuration.scala:109)
	at play.api.Configuration$.load(Configuration.scala:121)
	at play.api.Configuration$.load(Configuration.scala:128)
	at play.api.http.DefaultHttpErrorHandler$.setEditor$lzycompute(HttpErrorHandler.scala:487)
	at play.api.http.DefaultHttpErrorHandler$.setEditor(HttpErrorHandler.scala:486)
	at play.api.http.DefaultHttpErrorHandler$.onServerError(HttpErrorHandler.scala:495)
	at play.core.server.Server$$anonfun$handleErrors$1$1.applyOrElse(Server.scala:104)
	at play.core.server.Server$$anonfun$handleErrors$1$1.applyOrElse(Server.scala:100)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
Caused by: com.typesafe.config.ConfigException$Parse: application.conf @ file:./play27malformedconf/target/scala-2.12/classes/application.conf: 3: Expecting a value but got wrong token: '?' (Reserved character '?' is not allowed outside quotes) (if you intended '?' (Reserved character '?' is not allowed outside quotes) to be part of a key or string value, try enclosing the key or value in double quotes, or you may be able to rename the file .properties rather than .conf)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:201)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:197)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseValue(ConfigDocumentParser.java:251)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseObject(ConfigDocumentParser.java:475)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parse(ConfigDocumentParser.java:648)
	at com.typesafe.config.impl.ConfigDocumentParser.parse(ConfigDocumentParser.java:14)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:260)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:248)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:180)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:174)
[error] p.c.s.n.PlayRequestHandler - Exception caught in Netty
play.api.Configuration$$anon$1: Configuration error[application.conf @ file:./play27malformedconf/target/scala-2.12/classes/application.conf: 3: Expecting a value but got wrong token: '?' (Reserved character '?' is not allowed outside quotes) (if you intended '?' (Reserved character '?' is not allowed outside quotes) to be part of a key or string value, try enclosing the key or value in double quotes, or you may be able to rename the file .properties rather than .conf)]
	at play.api.Configuration$.configError(Configuration.scala:171)
	at play.api.Configuration$.load(Configuration.scala:109)
	at play.api.Configuration$.load(Configuration.scala:121)
	at play.api.Configuration$.load(Configuration.scala:128)
	at play.api.http.DefaultHttpErrorHandler$.setEditor$lzycompute(HttpErrorHandler.scala:487)
	at play.api.http.DefaultHttpErrorHandler$.setEditor(HttpErrorHandler.scala:486)
	at play.api.http.DefaultHttpErrorHandler$.onServerError(HttpErrorHandler.scala:495)
	at play.core.server.Server$$anonfun$handleErrors$1$1.applyOrElse(Server.scala:104)
	at play.core.server.Server$$anonfun$handleErrors$1$1.applyOrElse(Server.scala:100)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
Caused by: com.typesafe.config.ConfigException$Parse: application.conf @ file:./play27malformedconf/target/scala-2.12/classes/application.conf: 3: Expecting a value but got wrong token: '?' (Reserved character '?' is not allowed outside quotes) (if you intended '?' (Reserved character '?' is not allowed outside quotes) to be part of a key or string value, try enclosing the key or value in double quotes, or you may be able to rename the file .properties rather than .conf)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:201)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:197)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseValue(ConfigDocumentParser.java:251)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseObject(ConfigDocumentParser.java:475)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parse(ConfigDocumentParser.java:648)
	at com.typesafe.config.impl.ConfigDocumentParser.parse(ConfigDocumentParser.java:14)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:260)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:248)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:180)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:174)
[error] p.c.s.n.PlayRequestHandler - Exception caught in Netty
play.api.Configuration$$anon$1: Configuration error[application.conf @ file:./play27malformedconf/target/scala-2.12/classes/application.conf: 3: Expecting a value but got wrong token: '?' (Reserved character '?' is not allowed outside quotes) (if you intended '?' (Reserved character '?' is not allowed outside quotes) to be part of a key or string value, try enclosing the key or value in double quotes, or you may be able to rename the file .properties rather than .conf)]
	at play.api.Configuration$.configError(Configuration.scala:171)
	at play.api.Configuration$.load(Configuration.scala:109)
	at play.api.Configuration$.load(Configuration.scala:121)
	at play.api.Configuration$.load(Configuration.scala:128)
	at play.api.http.DefaultHttpErrorHandler$.setEditor$lzycompute(HttpErrorHandler.scala:487)
	at play.api.http.DefaultHttpErrorHandler$.setEditor(HttpErrorHandler.scala:486)
	at play.api.http.DefaultHttpErrorHandler$.onServerError(HttpErrorHandler.scala:495)
	at play.core.server.Server$$anonfun$handleErrors$1$1.applyOrElse(Server.scala:104)
	at play.core.server.Server$$anonfun$handleErrors$1$1.applyOrElse(Server.scala:100)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
Caused by: com.typesafe.config.ConfigException$Parse: application.conf @ file:./play27malformedconf/target/scala-2.12/classes/application.conf: 3: Expecting a value but got wrong token: '?' (Reserved character '?' is not allowed outside quotes) (if you intended '?' (Reserved character '?' is not allowed outside quotes) to be part of a key or string value, try enclosing the key or value in double quotes, or you may be able to rename the file .properties rather than .conf)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:201)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:197)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseValue(ConfigDocumentParser.java:251)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseObject(ConfigDocumentParser.java:475)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parse(ConfigDocumentParser.java:648)
	at com.typesafe.config.impl.ConfigDocumentParser.parse(ConfigDocumentParser.java:14)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:260)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:248)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:180)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:174)
```

</details>

--- a/core/play/src/main/scala/play/api/http/HttpErrorHandler.scala
+++ b/core/play/src/main/scala/play/api/http/HttpErrorHandler.scala
@@ -25,6 +25,9 @@ import play.utils.Reflect
 import scala.annotation.tailrec
 import scala.compat.java8.FutureConverters
 import scala.concurrent._
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
 import scala.util.control.NonFatal
 
 /**
@@ -499,10 +502,11 @@ class JsonHttpErrorHandler(environment: Environment, sourceMapper: Option[Source
  */
 object DefaultHttpErrorHandler
     extends DefaultHttpErrorHandler(HttpErrorConfig(showDevErrors = true, playEditor = None), None, None) {
-  private lazy val setEditor: Unit = {
-    val conf = Configuration.load(Environment.simple())
-    conf.getOptional[String]("play.editor").foreach(setPlayEditor)
-  }
+  private lazy val setEditor: Unit =
+    Try(Configuration.load(Environment.simple())) match {
+      case Success(conf) => conf.getOptional[String]("play.editor").foreach(setPlayEditor)
+      case Failure(_)    => // Very likely invalid config, not able to set the editor
+    }
 
   override def onClientError(request: RequestHeader, statusCode: Int, message: String): Future[Result] = {
     setEditor

--- a/core/play/src/main/scala/play/api/http/HttpErrorHandler.scala
+++ b/core/play/src/main/scala/play/api/http/HttpErrorHandler.scala
@@ -502,10 +502,15 @@ class JsonHttpErrorHandler(environment: Environment, sourceMapper: Option[Source
  */
 object DefaultHttpErrorHandler
     extends DefaultHttpErrorHandler(HttpErrorConfig(showDevErrors = true, playEditor = None), None, None) {
+  private val logger = Logger(getClass)
   private lazy val setEditor: Unit =
     Try(Configuration.load(Environment.simple())) match {
       case Success(conf) => conf.getOptional[String]("play.editor").foreach(setPlayEditor)
-      case Failure(_)    => // Very likely invalid config, not able to set the editor
+      case Failure(t) =>
+        logger.error(
+          "Can't read play.editor config because the configuration can't be loaded. This usually means there's a syntax error in your conf files.",
+          t
+        )
     }
 
   override def onClientError(request: RequestHeader, statusCode: Int, message: String): Future[Result] = {

